packages/sdk/src/protocols/filter.ts

```diff
@@ -345,7 +409,12 @@ class FilterSDK extends BaseProtocolSDK implements IFilterSDK {
       this.getActiveSubscription(pubsubTopic) ??
       this.setActiveSubscription(
         pubsubTopic,
-        new SubscriptionManager(pubsubTopic, this.connectedPeers, this.protocol)
+        new SubscriptionManager(
+          pubsubTopic,
+          this.protocol,
+          () => this.connectedPeers,
```

Comment:
ok, I think then we have `this.peers` changing a reference to some other array but we already passed a reference to the ctor. 

If we want to fight it:
- we should make sure no event kicks in and changes the reference;
- or use immutable array; 
- or reconstruct `SubscriptionManager` so that is is always in sync with peer (aka refer from `protocol`);

I don't think it worth the effort as we most likely want peers to be with the change of the event that happened. 
So the only option we can do is to refer to peers from some other link or keep getter for now.

Let's leave a comment with explanation if you decide to go with getter.

Reply:
can we use `this.protocol` for getting peers? I would go this direction

Reply:
so to simplify for `SubscriptionManager` 
```
 constructor(
    pubsubTopic: PubsubTopic,
    private protocol: FilterCore,
    // private peers: Peer[],
    private getPeers: () => Peer[],
    private readonly renewPeer: (peerToDisconnect: PeerId) => Promise<Peer>
  )
```

can be changed to 

```
 constructor(
    pubsubTopic: PubsubTopic,
    private protocol: FilterCore,
  )
```

Reply:
why not simply `FilterSDK`? 

```
 constructor(
    pubsubTopic: PubsubTopic,
    private protocol: FilterSDK,
  )
```